aten/src/ATen/native/SegmentReduce.cpp

```diff
@@ -116,9 +136,11 @@ Tensor segment_reduce_kernel(
   TORCH_CHECK(axis == 0, "Currently only dim=0 is supported!");
   TORCH_CHECK(data.dim() == 1);
   TORCH_CHECK(data.numel() > 0);
+  auto reduction_it = at::native::segmentReduce2REDUCE.find(reduce);
```

Comment:
I'm not sure using `std::map` is the best here, it's a heavyweight container. `scatter_reduce` uses just conditional statement (see `get_operator_enum`).